### ISSUE TITLE
Harmonize recipe steps to use snake-case and common-text-first format

### ISSUE THREAD

**Issue by mattwarkentin** [tidymodels/recipes#585]
This suggestion is in relation to https://github.com/tidymodels/recipes/pull/555#discussion_r506441161. I suggested naming `step_lmimpute()` to `step_impute_lm()`, with the possibility of future consolidation of the nomenclature for the imputation steps to use snake-case and the tidy format of placing common text first, to benefit from tab-autocompletion and fuzzy-narrowing.

I'd be happy to spin up a PR draft to soft-deprecate the existing imputation steps using `lifecycle`, if you'd like.

**Comment by juliasilge** [comment]
Thanks for your patience @mattwarkentin! 🙌 

We would be very happy for you to do a PR to make this change. A good example of how we are doing soft deprecation in recipes is [`step_downsample()`](https://github.com/tidymodels/recipes/blob/master/R/downsample.R).

**Comment by mattwarkentin** [comment]
Hi @juliasilge,

Sounds good. I'm on it. Just a few clarifying questions before I dig in. There will be a consistent deprecation pattern for all of the imputation steps, so I just want to make sure we agree on the approach.

- Should all internal/helper functions have their names changed accordingly (e.g. `tunable.step_knnimpute` to `tunable.step_impute_knn`, etc.)? I believe they should for consistency and clarity.

- Since we are only deprecating names, and not changing the functionality in a breaking way or removing the function altogether, for maintainability purposes, it would be nice to keep a single maintainable code base for each step so I was thinking a pattern something like:
```r
# New name
step_impute_* <- function(...) {
  # << implementation code >>
}

# Old name
step_*impute <- function(...) {
  # << deprecation code >>
  # call step_impute_*()
}
```
The deprecation message only appears when using the old name, of course, but still relies on current code. This way any updates to the new function are propagated to the old name, even as it is slowly deprecated out of service.

We can also have `?oldname` redirect to the documentation for `?newname`?

How does that sound?

**Comment by juliasilge** [comment]
Thanks for checking on these. 🙌 

- The internal helper functions should not be used by any external packages, so for clarity, **yes**, let's harmonize/change them to the new format, like `step_knnimpute_new()`.

- Yes, let's change the S3 class.

- That is how I would do the deprecation, changing the name of the function that implements the step and then calling it from the old name. Eventually, [it will look like this](https://github.com/ropensci/qualtRics/blob/master/R/deprecated.R#L27-L30).

- Let's redirect the documentation, and include a note in `@details` about when it changed:

> As of recipes 0.1.16, this function name changed from `step_blahimpute()` to `step_impute_blah()`.

**Comment by juliasilge** [comment]
Closed in #614
